mindscope_utilities/general_utilities.py

```diff
@@ -104,14 +104,22 @@ def event_triggered_response(data, t, y, event_times, t_before=1, t_after=1, out
     # set up a dictionary with key 'time' and
     # value as a linearly spaced time array
     step_size = 1/output_sampling_rate
-    data_dict = {
-        'time': np.linspace(
-            -t_before,
-            t_after,
-            int((t_before + t_after) / step_size + int(include_endpoint)),
-            endpoint=include_endpoint
-        )
-    }
+    # define a time array
+    n_steps = (t_before + t_after) / step_size
+    if n_steps != int(n_steps):
```

Comment:
while working on response_df noticed this too. Should include_endpoint still be an input variable? Because this would overwrite it without user's knowledge if this statement is True.

Reply:
I'm not quite sure I follow.

If we want an array from (0, 1) exclusive of the endpoint at 10 Hz, we would specify:
t_before = 0,
t_after = 1,
output_sampling_rate = 10,
include_endpoint=False
which would give:
[0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]

But if we want an array from (0, 1) inclusive of the endpoint at 10 Hz, we would specify:
t_before = 0,
t_after = 1,
output_sampling_rate = 10,
include_endpoint=True
which would give:
[0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]

Reply:
Sorry, I still struggle with typing clear messages. My concern is:
`include_endpoint` is an input variable, but it also gets automatically set to `True` in line 115 if `if n_steps != int(n_steps)` is `True`. So the scenario that I thought might be problematic is when someone calls this function, sets include_endpoint to False, but it then gets corrected to True because of the value of `n_step`. Maybe I am misunderstanding what the code does, but does that concern make sense?

Reply:
This is a fair point. What I'm doing is a little convoluted. Let me explain. Assume we want to use a sampling rate of 10Hz (step size = 0.1)
* in cases where the desired range is evenly divisible by the step size (say 0.0 to 1.0), all is well. We can use the `include_endpoint` that the user specified.
* In cases where the desired range is not evenly divisible by the step size (say 0.0 to 0.75), we have to make some adjustments. First, we round the number of desired steps down so that the range ends at 0.7. Then we need to set `include_endpoint` to True to ensure that we include 0.7. 

To hopefully make this a little easier to follow, I just broke all of this logic into its own function called `get_time_array`. I think it's actually nice to have this as its own function anyway because it could be a useful standalone method. It turns out that numpy is not good for this natively. For example, try running:
```python
import numpy as np
np.arange(-1, 0.75, 1/30)
```
You get back:
```python
array([-1.00000000e+00, -9.66666667e-01, -9.33333333e-01, -9.00000000e-01,
       -8.66666667e-01, -8.33333333e-01, -8.00000000e-01, -7.66666667e-01,
       -7.33333333e-01, -7.00000000e-01, -6.66666667e-01, -6.33333333e-01,
       -6.00000000e-01, -5.66666667e-01, -5.33333333e-01, -5.00000000e-01,
       -4.66666667e-01, -4.33333333e-01, -4.00000000e-01, -3.66666667e-01,
       -3.33333333e-01, -3.00000000e-01, -2.66666667e-01, -2.33333333e-01,
       -2.00000000e-01, -1.66666667e-01, -1.33333333e-01, -1.00000000e-01,
       -6.66666667e-02, -3.33333333e-02, -2.22044605e-16,  3.33333333e-02,
        6.66666667e-02,  1.00000000e-01,  1.33333333e-01,  1.66666667e-01,
        2.00000000e-01,  2.33333333e-01,  2.66666667e-01,  3.00000000e-01,
        3.33333333e-01,  3.66666667e-01,  4.00000000e-01,  4.33333333e-01,
        4.66666667e-01,  5.00000000e-01,  5.33333333e-01,  5.66666667e-01,
        6.00000000e-01,  6.33333333e-01,  6.66666667e-01,  7.00000000e-01,
        7.33333333e-01])
```
Notice that, due to decimal precision errors, the returned array has no actual 0.0 point. Instead, it has `-2.22044605e-16`. This is undesirable because it makes it impossible to search the array for the point where `t==0`

But, using our new function, we can do this:
```python
import mindscope_utilities
mindscope_utilities.get_time_array(
    t_start=-1, 
    t_end=0.75, 
    sampling_rate=30,
)
```
which returns:
```python
array([-1.        , -0.96666667, -0.93333333, -0.9       , -0.86666667,
       -0.83333333, -0.8       , -0.76666667, -0.73333333, -0.7       ,
       -0.66666667, -0.63333333, -0.6       , -0.56666667, -0.53333333,
       -0.5       , -0.46666667, -0.43333333, -0.4       , -0.36666667,
       -0.33333333, -0.3       , -0.26666667, -0.23333333, -0.2       ,
       -0.16666667, -0.13333333, -0.1       , -0.06666667, -0.03333333,
        0.        ,  0.03333333,  0.06666667,  0.1       ,  0.13333333,
        0.16666667,  0.2       ,  0.23333333,  0.26666667,  0.3       ,
        0.33333333,  0.36666667,  0.4       ,  0.43333333,  0.46666667,
        0.5       ,  0.53333333,  0.56666667,  0.6       ,  0.63333333,
        0.66666667,  0.7       ,  0.73333333])
```
Note that we have a True 0.0 now!